Small change to `derive_predictors_and_scores` for speed

PalkaPuri opened the pull request:
I noticed that generating a combined DF of predictors and scores was taking very long for large datasets (my kernel kept dying). This operation was very compute expensive because we were using `df.merge` which involves searching through column values to find matching rows. However, we can get away with using something simpler like `pd.concat` since all predictor/score DataFrames inherit the grid from `local_windows` and hence the rows match by design. This should help clear the speed bottleneck

PalkaPuri commented:
Added option to scale values based on empirical CDF

rfl-urbaniak reviewed (changes requested):
Please pull origin from the current version of staging, make sure you resolve all conflicts and pass all the tests.

PalkaPuri commented:
@rfl-urbaniak @dimkab  Just finished going over this branch. The following changes were implemented: 
1. final `derivedDF` is now generated by concatenating each predictor/score DF on index instead of using `df.merge` which was more computationally intensive
2. Added time logging for generation of `derivedDF` and `local_windows`
3. Updated the UserWarning in case of NaNs : previously we said X *frames* are dropped from derivedDF. But that is incorrect as the number X corresponds to **rows** of the df (which may be coming from any number of frames of the data). 
4. Updated `add_scaled_scores` option to scale values according to the empirical CDF. Previously scaling using min/max did not do anything as all predictors/scores are already scaled to be between 0 and 1.
5. The output of the corresponding test notebook was changed due to these updates

Note: After merging these changes we would need to run all the doc notebooks again, as 2,3, and 4 can potentially change the outputs of the cells. I did not do this just yet so as to not overwhelm you both with 100 file changes in one PR.

rfl-urbaniak reviewed (changes requested):
I am not convinced that replacing merge with concat has no strange consequences at least for some of the predictors. As a sanity check I re-run communicators inference with your proposed modification and the posterior marginals are significantly different from the ones we currently have. I think this modification, if indeed correct, needs additional explicit tests that ensure proper functioning, involving predictors other than velocity too.
